src/__tests__/extended-id.test.ts

```diff
@@ -37,3 +37,88 @@ describe('pattern matcher', () => {
     })
   })
 })
+
+describe('interpretIdentifier', () => {
+  const pubKey = '0x02b97c30de767f084ce3080168ee293053ba33b235d7116a3263d29f1450936b71'
+  const checksumAddress = '0xC662e6c5F91B9FcD22D7FcafC80Cf8b640aed247'
+
+  it('parses ethereumAddress', () => {
+    const { address, publicKey, network } = interpretIdentifier(checksumAddress.toLowerCase())
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toBeUndefined()
+  })
+  it('parses ethereumAddress with checksum', () => {
+    const { address, publicKey, network } = interpretIdentifier(checksumAddress)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toBeUndefined()
+  })
+  it('parses did:ethr with address', () => {
+    const { address, publicKey, network } = interpretIdentifier(`did:ethr:${checksumAddress}`)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toBeUndefined()
+  })
+  it('parses did:ethr with address and version', () => {
+    const { address, publicKey, network } = interpretIdentifier(`did:ethr:${checksumAddress}?versionId=42`)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toBeUndefined()
+  })
+  it('parses did:ethr with address and network', () => {
+    const { address, publicKey, network } = interpretIdentifier(`did:ethr:0x1:${checksumAddress}`)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toEqual('0x1')
+  })
+  it('parses did:ethr with address and sub-network', () => {
+    const { address, publicKey, network } = interpretIdentifier(`did:ethr:rsk:testnet:${checksumAddress}`)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toEqual('rsk:testnet')
+  })
+  it('parses did:ethr with address and sub-network and version', () => {
+    const { address, publicKey, network } = interpretIdentifier(`did:ethr:rsk:testnet:${checksumAddress}?versionId=42`)
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
```

Comment:
well, hard to check whether this is correct.

Reply:
what do you mean?
what is at risk of not being correct?